self_test/generic/complex_matrices_test/cc_complex_matrix_test.cc

```diff
@@ -0,0 +1,165 @@
+// LIC// ====================================================================
+// LIC// This file forms part of oomph-lib, the object-oriented,
+// LIC// multi-physics finite-element library, available
+// LIC// at http://www.oomph-lib.org.
+// LIC//
+// LIC// Copyright (C) 2006-2021 Matthias Heil and Andrew Hazel
+// LIC//
+// LIC// This library is free software; you can redistribute it and/or
+// LIC// modify it under the terms of the GNU Lesser General Public
+// LIC// License as published by the Free Software Foundation; either
+// LIC// version 2.1 of the License, or (at your option) any later version.
+// LIC//
+// LIC// This library is distributed in the hope that it will be useful,
+// LIC// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// LIC// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// LIC// Lesser General Public License for more details.
+// LIC//
+// LIC// You should have received a copy of the GNU Lesser General Public
+// LIC// License along with this library; if not, write to the Free Software
+// LIC// Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+// LIC// 02110-1301  USA.
+// LIC//
+// LIC// The authors may be contacted at oomph-lib@maths.man.ac.uk.
+// LIC//
+// LIC//====================================================================
+
+// Oomph-lib includes
+#include "generic.h"
+
+using namespace std;
+using namespace oomph;
+
+void print_complex_vector(Vector<complex<double>>& x)
+{
+  unsigned vector_length = x.size();
+  for (unsigned i = 0; i < vector_length; i++)
+  {
+    cout << x[i] << endl;
+  }
+}
+
+void print_cc_complex_matrix(CCComplexMatrix& matrix)
+{
+  unsigned n_row = matrix.nrow();
+  unsigned n_col = matrix.ncol();
+  for (unsigned i = 0; i < n_row; i++)
+  {
+    for (unsigned j = 0; j < n_col; j++)
+    {
+      cout << matrix(i, j) << ", ";
+    }
+    cout << endl;
+  }
+}
+
+int main()
+{
+  // test default constructor
+  CCComplexMatrix matrix_default;
+  cout << matrix_default.nrow() << endl;
+  cout << matrix_default.ncol() << endl;
+
+  constexpr unsigned long n_col_rect = 2;
+  constexpr unsigned long n_row_rect = 4;
+  // Number of non-zero entries
+  constexpr unsigned n_nz_rect = 4;
```

Comment:
I included the `long` to match the interface for the class constructor. It may not be strictly necessary but I added it to make it clear for anyone looking through the tests for an example use case.
The number of non-zeros is not used in the constructor, but is used as an `int` in the `column_start` Vector. So I defaulted to just an `unsigned`.
Did you want me to change anything?

Reply:
Ahh, I see! Then feel free to leave it as is.